src/middleware.ts

```diff
@@ -106,3 +107,84 @@ export const withWebhookSignatureVerified =
 
     return handler(request);
   };
+
+export interface DashboardTokenPayload extends JwtPayload {
+  app: string;
+}
+
+export const withJWTVerified =
+  (getAppId: (request: Request) => Promise<string | undefined>): Middleware =>
+  (handler) =>
+  async (request) => {
+    const { [SALEOR_DOMAIN_HEADER]: saleorDomain, "authorization-bearer": token } = request.headers;
+
+    if (token === undefined) {
+      return Response.BadRequest({
+        success: false,
+        message: "Missing token.",
+      });
+    }
+
+    let tokenClaims;
+    try {
+      tokenClaims = jwt.decode(token as string);
+    } catch (e) {
+      console.error(e);
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    if (tokenClaims === null) {
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    if ((tokenClaims as DashboardTokenPayload).iss !== saleorDomain) {
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
```

Comment:
Also being explicit. We can say that token domain doesn't match or something like this?